packages/core/useReducer/ComplexDemo.tsx

```diff
@@ -0,0 +1,148 @@
+import 'vue-tsx-support/enable-check'
+import Vue from 'vue'
+import { defineComponent, provide, inject } from 'vue-demi'
+import { useReducer } from '.'
+
+const initialState: ComplexState = {
+  loading: false,
+  data: [],
+}
+type ComplexState = {
+  loading: boolean
+  data: number[]
+  error?: Error
+}
+
+const fetchMock = () =>
+  new Promise((resolve, reject) => {
+    window.setTimeout(() => {
+      if (Math.random() > 0.5) resolve([1, 2, 3, 4, 5])
+      else reject(new Error('Request Error'))
+    }, 1500)
+  })
+
+function reducer(state: any, action: any) {
+  switch (action.type) {
+    case 'loadData':
+      state.error = undefined
+      state.loading = true
+      return fetchMock()
+        .then(data => (state.data = data))
+        .catch(e => (state.error = e))
+        .finally(() => (state.loading = false))
+    case 'addItem':
+      state.data.push(action.payload)
+      return
+    case 'removeItem':
+      state.data = state.data.filter((ele: number) => ele !== action.payload)
+  }
+}
+
+const ErrorUi = defineComponent({
+  props: {
+    message: String,
+  },
+  setup() {
+    const dispatch = inject<Function>('dispatch', () => {})
+    const triggerReload = () => dispatch({ type: 'loadData' })
+
+    return { triggerReload }
+  },
+
+  render(this: Vue & { triggerReload: () => void }) {
+    return (
+      <div>
+        {this.$props.message}
+        <button onClick={() => this.triggerReload()}>Reload</button>
+      </div>
+    )
+  },
+})
+
+const DataUi = defineComponent({
+  props: {
+    data: {
+      type: Array,
+      default: () => [],
+    },
+  },
+  render(this: Vue) {
+    return (
+      <ul>
+        {this.$props.data.map((ele: number) => (
+          <ListItem id={ele} />
```

Comment:
`##[error]packages/core/useReducer/ComplexDemo.tsx(73,21): error TS2769: No overload matches this call.`
I dont understand how i can solve this.